src/oscar/apps/es_search/facet.py

```diff
@@ -0,0 +1,185 @@
+from collections import OrderedDict
+from purl import URL
+
+from django.utils.translation import gettext_lazy as _
+
+
+class Facet(object):
+
+    facet_name = None
+
+    def __init__(self, name, agg, request_url, user_defined_facets, selected_facets={}):
```

Comment:
Please drop `{}` as a default argument, as this is passed by reference and is mutable - i.e., every time a facet is initialised the **same** dictionary will be used. Need to default to `None` and then assign a dictionary in the method itself.